src/QsCompiler/LlvmBindings/Metadata/MDNode.cs

```diff
@@ -0,0 +1,136 @@
+        /// <param name="index">Index of the operand</param>
+        /// <returns>Operand</returns>
+        /// <exception cref="InvalidCastException">When the operand is not castable to <typeparamref name="T"/></exception>
+        /// <exception cref="ArgumentOutOfRangeException">When the index is out of range for the operands of this node</exception>
+        public T? GetOperand<T>( int index )
+            where T : LlvmMetadata
+        {
+            return this.Operands.GetOperand<T>( index );
+        }
+
+        public Value? GetOperandValue( int index )
+        {
+            return this.Operands.GetOperandValue(index);
+        }
+
+        /// <summary>Gets a string operand by index</summary>
+        /// <param name="index">Index of the operand</param>
+        /// <returns>String value of the operand</returns>
+        public string GetOperandString( int index )
+            => GetOperand<MDString>( index )?.ToString( ) ?? string.Empty;
+
+        /* TODO: Consider adding these additional properties/methods
+        public bool IsTBAAVTableAccess { get; }
+
+        public TempMDNode Clone() {...}
+
+        public void ReplaceOperandWith(unsigned i, LlvmMetadata other) {...}
+        public static MDNode Concat(MDNode a, MDNode b) {...}
+        public static MDNode Intersect(MDNode a, MDNode b) {...}
+        public static MDNode GetMostGenericTBAA(MDNode a, MDNode b) {...}
+        public static MDNode GetMostGenericFPMath(MDNode a, MDNode b) {...}
+        public static MDNode GetMostGenericRange(MDNode a, MDNode b) {...}
+        public static MDNode GetMostGenericAliasScope(MDNode a, MDNode b) {...}
+        public static MDNode GetModtGenericAlignmentOrDereferenceable(MDNode a, MDNode b) {...}
+
+        public static T ReplaceWithPermanent<T>() where T:MDNode
+        public static T ReplaceWithUniqued<T>() where T:MDNode
+        public static T ReplaceWithDistinct<T>() where T:MDNode
+        public static DeleteTemporary(MDNode node) {...}
+        */
+
+        internal MDNode( LLVMMetadataRef handle )
+            : base( handle )
+        {
+            Operands = new MetadataOperandCollection( this );
+        }
+
+        [SuppressMessage( "Reliability", "CA2000:Dispose objects before losing scope", Justification = "Context created here is owned, and disposed of via the ContextCache" )]
+        internal static T? FromHandle<T>( LLVMMetadataRef handle )
+            where T : MDNode
+        {
+            return handle == default ? null : FromHandle<T>( GetMetadataContext( handle ), handle );
+        }
+
+        private void ThrowIfDeleted( )
+        {
+            if( IsDeleted )
+            {
+                throw new InvalidOperationException( "Cannot operate on a deleted node" );
+            }
+        }
+
+        private static Context GetMetadataContext( LLVMMetadataRef metadataHandle )
+        {
+            // TODO (Q#): we currently expect exactly one context, since we have no way
```

Comment:
Glad to see this called out in code! I don't think this needs to be a "TODO" since as of now there isn't pending work associated with this, so that prefix can just be removed and the comment text kept as-is.

Reply:
Will do! Do you think it's worth keeping the `Q#` prefix to make it clear that the comment wasn't imported from Ubiquity?

Reply:
That's a good idea; the Q# prefix will clarify origin!

Reply:
https://github.com/microsoft/qsharp-compiler/pull/1026/commits/6f0de6f347cf1b872daa29dc25d9e466d645288f